### ISSUE TITLE
% switches to insert mode

### ISSUE THREAD

**Issue by mgarort** [airblade/vim-matchquote#7]
Hello,

In my configuration, using `%` on any character deletes the current line, switches to insert mode and inserts part of a mapping from vim-matchquote. I am not sure what could be causing this, but to be fair I have a relatively convoluted configuration.

For example, pressing `%` in normal mode on the line

```vimscript
Plugin 'airblade/vim-matchquote'
```
mangles it to 
```vimscript
l <SNR>39_matchquote('n')
```

and leaves the cursor in insert mode.

Any ideas what could be causing this?

Thanks in advance! 

(And thanks for writing this plugin as well. I think this kind of functionality should be built in.)

**Comment by airblade** [comment]
> using % on any character deletes the current line, switches to insert mode and inserts part of a mapping from vim-matchquote

I think this can be filed under unintended behaviour ;)

Has it just started happening?

Assuming for now that no other plugin is involved... the only tricky part of this plugin is invoking the matchit plugin if the cursor isn't on one of our quote characters.  It's tricky because matchit's "API" has changed several times over the years.

To find out which version we're dealing with, please could you let me know what this produces?

```
vim --cmd 'let g:loaded_matchquote=1' -c 'echo maparg("%", "n")'
```

Also it might be handy to see if you can reproduce the problem with only this plugin installed.

**Comment by mgarort** [comment]
Hi,

Sorry for the delay, work has kept me busy during the last days.

I found what was causing the issue: the mappings defined by vim-matchquote are recursive, so any interfering mappings may result in the RHS being mapped to something else and causing unintended behaviour. In my case, what was being remapped was the colon `:`, since I have remapped `:` to `,` (so that we can repeat `f`, `F,` `t` and `T` motions with `;` and reverse motions with `<S-;>`, i.e. with `:`), and `<Space>` to `:`.

So the issue is fixed by changing the following mappings to the `noremap` versions:

```
nnoremap <silent> %      :call <SID>matchquote('n')<CR>
xnoremap <silent> % :<C-U>call <SID>matchquote('x')<CR>
```
